Machine_Learning/notebooks/czmodel/Regresssion_PyTorch_5_0_0.ipynb

```diff
@@ -0,0 +1,669 @@
+    "# Regression labels contain a real number for each pixel\n",
+    "LABELS_FOLDER = 'Mouse_Kidney_images/labels/'\n",
+    "\n",
+    "# Path to the data on GitHub\n",
+    "GITHUB_TRAINING_DATA_PATH = 'https://raw.githubusercontent.com/zeiss-microscopy/OAD/master/Machine_Learning/notebooks/czmodel/Mouse_Kidney_images.zip'\n",
+    "GITHUB_MODEL_CONVERSION_SPEC_PATH = 'https://raw.githubusercontent.com/zeiss-microscopy/OAD/master/Machine_Learning/notebooks/czmodel/regression_conversion_spec_pytorch.json'\n",
+    "\n",
+    "# Define the number of input color channels\n",
+    "# This means that the inputs are grayscale with one channel only\n",
+    "CHANNELS = 1 \n",
+    "\n",
+    "# The size of image crops to train the model with\n",
+    "CROP_SIZE = 512"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "0JUvTmdVrEEF"
+   },
+   "source": [
+    "### Download data if it's not available on disk\n",
+    "If this notebook is run e.g. as a colab notebook, it does not have access to the data folder on gitub via disk access. \n",
+    "In that case we need to download the data from github first."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "LKeWmEpSrEEF"
+   },
+   "outputs": [],
+   "source": [
+    "import requests\n",
+    "\n",
+    "# Download training data\n",
+    "if not (os.path.isdir(IMAGES_FOLDER) and os.path.isdir(LABELS_FOLDER)):\n",
+    "    compressed_data = './Mouse_Kidney_images.zip'\n",
+    "    if not os.path.isfile(compressed_data):\n",
+    "        import io\n",
+    "        response = requests.get(GITHUB_TRAINING_DATA_PATH, stream=True)\n",
+    "        compressed_data = io.BytesIO(response.content)\n",
+    "        \n",
+    "    import zipfile\n",
+    "    with zipfile.ZipFile(compressed_data, 'r') as zip_accessor:\n",
+    "        zip_accessor.extractall('./')\n",
+    "        \n",
+    "# Download model conversion spec\n",
+    "if not os.path.isfile('regression_conversion_spec_pytorch.json'):\n",
+    "    response = requests.get(GITHUB_MODEL_CONVERSION_SPEC_PATH, stream=True)\n",
+    "    with open('regression_conversion_spec_pytorch.json', 'wb') as handle:\n",
+    "        handle.write(response.content)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "h9B-lhwXrEEG"
+   },
+   "source": [
+    "### Define a PyTorch dataset to pre-process the images\n",
+    "Since the dataset contains very large images we need to train on smaller crops in order to not exhaust the GPU memory"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "hrIPBmrkrEEG"
+   },
+   "outputs": [],
+   "source": [
+    "from torch.utils.data import DataLoader, Dataset\n",
+    "import cv2\n",
+    "\n",
+    "# Dataset class\n",
+    "class SampleDataset(Dataset):\n",
+    "    def __init__(self, images_folder:str, labels_folder:str, transforms=None):\n",
```

Comment:
We partially have type hints in the code. It would be great if we could be consistend w.r.t. using type hints. It's up to you to use them. I would also be fine with not using any type hints in a script like that, but having them would IMO be cleaner.